File: news_manager_web/src/main/webapp/js/controller/categoryController.js
app.controller('categoryController',function ($scope,$controller, categoryService) {
    $controller('baseController',{$scope:$scope})

    $scope.init = function () {
        $scope.findPage();
    }

    $scope.findPage = function () {
        if(!$scope.keywords){
            $scope.keywords=''
        }
        categoryService.findPage($scope.paginationConf.currentPage,$scope.paginationConf.itemsPerPage,$scope.keywords).success(function (reponse) {
            $scope.list = reponse.rows;
            $scope.paginationConf.totalItems = reponse.total;
        });
    }

    $scope.update = function () {
        var object;
        if($scope.entity.id){
            object = categoryService.update($scope.entity);
        }else {
            object = categoryService.add($scope.entity);
        }
        object.success(function (response) {
            if(response.success){
                $scope.findPage();
            }else {
                alert(response.msg);
            }
        });
    }

    $scope.findOne = function (id) {
        categoryService.findOne(id).success(function (response) {
            $scope.entity = response;
        });
    }

    $scope.delete = function (id) {
        var idList=[]
        if(id){
            idList.push(id);
        }else {
            idList = $scope.ids;
        }
        categoryService.delete(idList).success(function (response) {
            $scope.ids=[];
            if(response.success){
                $scope.findPage();
            }else{
                alert(response.msg);
            }
        })
    }

});

File: news_manager_web/src/main/java/cn/hncu/controller/CategoryController.java
package cn.hncu.controller;

import cn.hncu.entity.PageResult;
import cn.hncu.entity.Result;
import cn.hncu.pojo.Category;
import cn.hncu.service.ICategoryService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * Created by Enzo Cotter on 2019/5/15.
 */
@RestController
@RequestMapping("/category")
public class CategoryController {

    @Autowired
    private ICategoryService categoryService;

    @RequestMapping("/findPage")
    public PageResult findPage(int currentPage,int pageSize,String keywords){
        PageResult page = categoryService.findPage(currentPage, pageSize,keywords);
        return page;
    }

    @RequestMapping("/findAll")
    public List<Category> findAll(){

        return categoryService.findAll();
    }

    @RequestMapping("/findOne")
    public Category findOne(Integer id){
        return categoryService.findOne(id);
    }

    @RequestMapping("/add")
    public Result add(@RequestBody Category category){
        try {
            categoryService.add(category);
            return new Result(true,"增加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"新增失败");
        }
    }

    @RequestMapping("/update")
    public Result update(@RequestBody Category category){
        try {
            categoryService.update(category);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    @RequestMapping("/delete")
    public Result delete(Integer[] ids){
        try {
            categoryService.delete(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }
}
